java/core/src/test/java/com/whylogs/core/TestColumnProfile.java

```diff
@@ -0,0 +1,144 @@
+package com.whylogs.core;
+
+import com.whylogs.core.metrics.IntegralMetric;
+import com.whylogs.core.metrics.MetricConfig;
+import com.whylogs.core.metrics.components.MaxIntegralComponent;
+import com.whylogs.core.metrics.components.MinIntegralComponent;
+import com.whylogs.core.resolvers.StandardResolver;
+import com.whylogs.core.schemas.ColumnSchema;
+import com.whylogs.core.views.ColumnProfileView;
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+@Test
+public class TestColumnProfile {
+  private String columnName = "testColumn";
+  private int CACHE_SIZE = 2;
+
+  private ColumnProfile<Integer> getDefaultColumnProfile() {
+    ColumnSchema standardSchema =
+        new ColumnSchema(Integer.class, new MetricConfig(), new StandardResolver());
+    ColumnProfile<Integer> result = new ColumnProfile<>(columnName, standardSchema, CACHE_SIZE);
+    result.addMetric(IntegralMetric.zero(new MetricConfig()));
+    return result;
+  }
+
+  @Test
+  public void testColumnProfileInit() {
+    ColumnProfile<Integer> profile = getDefaultColumnProfile();
+    Assert.assertEquals(profile.getName(), columnName);
+    Assert.assertEquals(profile.getSchema().getType(), Integer.class);
+    Assert.assertEquals(profile.getCachedSize(), CACHE_SIZE);
+  }
+
+  @Test
+  public void testAddMetric() {
+    ColumnProfile<Integer> profile = getDefaultColumnProfile();
+    profile.addMetric(IntegralMetric.zero(new MetricConfig()));
```

Comment:
The column already has an `IntegralMetric`. I think the `addMetric()` behavior in that case is to replace the old metric instance & log a warning. It looks like you test that with `metric2` below, so maybe skip the `addMetric()` here?